mat64/dense.go

```diff
@@ -164,13 +175,53 @@ func (m *Dense) View(i, j, r, c int) Matrix {
 	t.mat.Data = t.mat.Data[i*t.mat.Stride+j : (i+r-1)*t.mat.Stride+(j+c)]
 	t.mat.Rows = r
 	t.mat.Cols = c
+	t.capRows -= i
+	t.capCols -= j
 	return &t
 }
 
+func (m *Dense) Grow(r, c int) {
+	if r < 0 || c < 0 {
+		panic(ErrIndexOutOfRange)
+	}
+	if r == 0 || c == 0 {
+		return
+	}
+
+	r += m.mat.Rows
+	c += m.mat.Cols
+
+	switch {
+	case m.mat.Rows == 0 || m.mat.Cols == 0:
+		m.mat = blas64.General{
+			Rows:   r,
+			Cols:   c,
+			Stride: c,
+			Data:   use(m.mat.Data, r*c),
+		}
+	case r > m.capRows || c > m.capCols:
+		t := Dense{mat: blas64.General{
```

Comment:
Doesn't look like the note got through:

On the flipside, if you grow it, it doesn’t decrease in size, and then you expand it again, you may be surprised to find you have zeros rather than the previous values.

This is a pretty bizarre case. If I have a 4x4 matrix, I view it to have a 4x3 matrix. I Grow an extra row to that, causing an allocation and giving a 5x3 matrix. My personal assumption would be that while I previously stated it had 4 rows, I voided that contract when I added an extra 3-sized row. This meshes with the idea that when I Grow it to get 4 rows, I don’t see the old values I had when it was a 4x4 matrix — new data was allocated. 

That would be my two cents, but it isn’t clear. I hope there isn’t real code that actually relies on this behavior.